code/modules/research/xenoarchaeology/artifact/artifact_duplicator.dm

```diff
@@ -0,0 +1,188 @@
+/obj/machinery/duplicator
+	name = "alien machine"
+	desc = "It's some kind of large computer. There appear to be disks of some kind anchored to the sides."
+	icon = 'icons/obj/xenoarchaeology.dmi'
+	icon_state = "duplicator_idle"
+	density = 1
+	var/obj/structure/duplicator_pod/leftpod
+	var/obj/structure/duplicator_pod/rightpod
+	var/deployed = FALSE
+	var/duplicating = FALSE
+	var/dupe_time = 10 SECONDS
+
+	idle_power_usage = 100
+	active_power_usage = 1000
+	use_power = 1
+
+	machine_flags = WRENCHMOVE | FIXED2WORK
+
+/obj/machinery/duplicator/New()
+	..()
+	leftpod = new(src)
+	rightpod = new(src)
+	update_icon()
+
+/obj/machinery/duplicator/update_icon()
+	..()
+	icon_state = initial(icon_state)
+	if(stat & NOPOWER || !anchored)
+		icon_state = "duplicator_off"
+	else if(duplicating)
+		icon_state = "duplicator_duplicating"
+
+/obj/machinery/duplicator/power_change()
+	. = ..()
+	update_icon()
+	if(anchored && !deployed)
+		deploy_pods()
+	else if(!anchored && deployed)
+		retract_pods()
+
+/obj/machinery/duplicator/proc/deploy_pods()
+	desc = "It's some kind of large computer."
+	visible_message("The disks detach from the sides of \the [src] and unfold into large pods.")
+	leftpod.forceMove(locate(x-1,y,z))
```

Comment:
Use `get_step()`, and have it be based on direction variables on the object - it would be useful to be able to map these for vaults where they aren't left-to-right.